ros2cli/ros2cli/daemon/__init__.py

```diff
@@ -153,7 +171,11 @@ def main(*, argv=None):
     assert args.rmw_implementation == rclpy.get_rmw_implementation_identifier()
     assert args.ros_domain_id == get_ros_domain_id()
 
-    serve(make_xmlrpc_server(), timeout=args.timeout)
+    server = make_xmlrpc_server()
+    try:
+        serve(server, timeout=args.timeout)
+    finally:
+        server.server_close()
```

Comment:
Pretty sure `LocalXMLRPCServer` is a context manager.

```suggestion
    with make_xmlrpc_server() as server:
        serve(server, timeout=args.timeout)
```